single-function/valid-parentheses/submission.js

```diff
@@ -0,0 +1,48 @@
+const isValidParentheses = (str) => {
+  if(str.length <= 1) { return false }
+  const parenMap = { '{':'}', '(':')', '[':']' };
+  const stack = [];
+  for (let i = 0; i < str.length; i++) {
+    if (parenMap[str[i]]) {
+      stack.push(parenMap[str[i]]);
+    } else if (stack.pop() !== str[i]) {
+      return false;
+    }
+  }
+  return (str.length % 2 !== 0) || (stack.length !== 0) || true;
+};
+
+/*
+Input: '()()()'
+Output: boolean
+Constraints: n/a
+
+Strategy: Step through the given string, if a valid opening paren push closing paren on to stack.
```

Comment:
"Step through" -> "Iterate".  Let's get them to use just a small set of standard terminology. Otherwise they might next say "Go through" etc.

Reply:
Let's boil down the strategy a bit more to: Each open paren adds to a stack of expected closing parens. Each closing paren must match its expected value, popped from the stack.